python/cudf/cudf/core/dataframe.py

```diff
@@ -371,9 +371,43 @@ def _setitem_tuple_arg(self, key, value):
                 )
             self._frame._data.insert(key[1], new_ser._column)
         else:
-            if is_scalar(value):
-                for col in columns_df._column_names:
-                    self._frame[col].loc[key[0]] = value
+            if is_scalar(value) and not isinstance(
+                self._frame.index, cudf.MultiIndex
+            ):
+                try:
+                    if len(columns_df._column_names) > 0:
+                        self._frame[
+                            columns_df._column_names[0]
+                        ].loc._loc_to_iloc(key[0])
+                    for col in columns_df._column_names:
+                        self._frame[col].loc[key[0]] = value
+                except KeyError:
+                    idx = self._frame.index
+                    if isinstance(idx, cudf.RangeIndex):
+                        if isinstance(key[0], int) and (
+                            key[0] == idx[-1] + idx.step
+                        ):
+                            idx_copy = cudf.RangeIndex(
+                                start=idx.start,
+                                stop=idx.stop + idx.step,
+                                step=idx.step,
+                                name=idx.name,
+                            )
+                        else:
+                            idx_copy = idx._as_int_index()
+                            _append_new_row_inplace(idx_copy._column, key[0])
+                    else:
+                        # TODO: Modifying index in place is bad because
+                        # our index are immutable, but columns are not (which
+                        # means our index are mutable with internal APIs).
+                        # Get rid of the deep copy once columns too are
+                        # immutable.
+                        idx_copy = idx.copy(deep=True)
+                        _append_new_row_inplace(idx_copy._column, key[0])
+
+                    for col in columns_df._column_names:
+                        _append_new_row_inplace(self._frame._data[col], value)
+                    self._frame._index = idx_copy
```

Comment:
Is there a way to define this in `_FrameIndexer.__setitem__` so it can be used by `Series` and `DataFrame`?